Assets/MixedRealityToolkit/_Core/Devices/BaseController.cs

```diff
@@ -141,14 +141,7 @@ public void SetupConfiguration(Type controllerType)
         /// <param name="mappings">Configured mappings from a controller mapping profile</param>
         public void AssignControllerMappings(MixedRealityInteractionMapping[] mappings)
         {
-            var interactions = new MixedRealityInteractionMapping[mappings.Length];
-
-            for (uint i = 0; i < mappings.Length; i++)
-            {
-                interactions[i] = new MixedRealityInteractionMapping(i, mappings[i].Description, mappings[i].AxisType, mappings[i].InputType, mappings[i].MixedRealityInputAction);
-            }
-
-            Interactions = interactions;
+            Interactions = mappings;
```

Comment:
Should we make sure the controller isn't being used/updated while we update the mappings?

Reply:
Not needed, as the Assign function is only called on initialisation atm.

Reply:
Seeing as this is a public method we might see ppl try to set it at runtime which might cause issues.